datacube/config.py

```diff
@@ -92,6 +94,11 @@ def __init__(self, config: configparser.ConfigParser,
         if env:
             if config.has_section(env):
                 self._env = env
+                if not re.match(r'[a-z]+$', self._env):
```

Comment:
This is different from the regex in `check_valid_env_name` in 1.9: `re.fullmatch(r"^[a-z][a-z0-9]*$", name)`

Reply:
Oops, different from the description in the error message too (which does match 1.9) - thanks.